runtime/interpreter/errors.go

```diff
@@ -250,6 +250,10 @@ func (e DereferenceError) Error() string {
 	return "dereference failed"
 }
 
+func (e DereferenceError) SecondaryError() string {
+	return "the value being referenced has been destroyed or moved"
```

Comment:
This error is also reported when a storage reference is dereferenced and the value stored at the path has the wrong type, or there is no value stored at the path